compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -462,6 +464,15 @@ pub fn make_format_args(
             let msg = if let FormatArgumentKind::Named(_) = args.explicit_args()[i].kind {
                 "named argument never used"
             } else {
+                if let Some(expr) = args.explicit_args()[i].expr.to_ty() {
+                    if let Some(symbol) = expr.kind.is_simple_path() {
+                        let current_arg = symbol.as_str().to_owned();
+                        let current_arg_ph = format!("{{{current_arg}}}");
+                        if current_arg.len() > 0 && fmt_str.contains(current_arg_ph.as_str()) {
```

Comment:
This is a bit brittle, isn't it? It would trigger in the face of a formatting string like `format!("{{x}}", x)`. Instead, you might want to look at `pieces`, which holds the parsed format string or, more likely, `template` which holds the fully parsed result and look at the `argument` of every `FormatArgsPiece::Placeholder`.